pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -266,7 +266,7 @@ private void updateInstanceConfigInHelix(Map<String, String> props) {
     _helixAdmin.setConfig(scope, props);
   }
 
-  private void addInstanceTagIfNeeded(String clusterName, String instanceName) {
+  private void updateInstanceConfigIfNeeded(String clusterName, String instanceName, Configuration serverConf) {
     InstanceConfig instanceConfig = _helixAdmin.getInstanceConfig(clusterName, instanceName);
     List<String> instanceTags = instanceConfig.getTags();
     if (instanceTags == null || instanceTags.size() == 0) {
```

Comment:
` _helixAdmin.addInstanceTag()` can be simplified to `instanceTags.add()`

Reply:
_helixAdmin.addInstanceTag() writes to zk. InstanceTag.add() does not do that -- instanceTag is merely a reference. I do not think the former can be simplified to latter. Did I miss anything here?

Reply:
That is why I suggested adding a flag to track whether we need to update and write the instance config back to ZK. Each `_helixAdmin.addInstanceTag()` reads the instance config and writes it back, which can be saved with the flag. Overall we should only read the instance config once, and write it back once if it needs to be updated.

Reply:
Got it. Revised.